mslib/msui/mscolab.py

```diff
@@ -1977,6 +1978,9 @@ def handle_import_msc(self, file_path, extension, function, pickertype):
                 model = ft.WaypointsTableModel(waypoints=new_waypoints)
                 xml_doc = self.waypoints_model.get_xml_doc()
                 xml_content = xml_doc.toprettyxml(indent="  ", newl="\n")
+            if not verify_waypoint_data(xml_content):
+                show_popup(self.ui, "Import Failed", f"The file - {file_name}, was not imported!", 0)
+                return
```

Comment:
If the intention is to protect against e.g. maliciously crafted XML or the like, then verification must be the first thing that is done, before using the XML for anything else. In this case though, the XML might be used already here: https://github.com/Open-MSS/MSS/blob/ff744f5f4dadf09407e81688a0da18b1cd771d3e/mslib/msui/mscolab.py#L1971

Reply:
Also, if `function` is not None, then that supplied function would need to do the verification as well. But this is a theoretical concern, it seems like for XML files it is always None, and it is only used to provide a way to read CSV files instead.